src/views/Pathways/Courses/form/course.js

```diff
@@ -0,0 +1,42 @@
+import * as yup from "yup"
+
+const createdAtField = {
+  name: "createdAt",
+  type: "date",
+  validation: yup.date().typeError("required"),
+  customProps: {
+    label: "Created at",
+    variant: "dialog",
+    disabled: true,
+  },
+}
+
+export const getCourseAddFormStructure = (course,allCourses) => {
+
+  const optionsList=[]
+  let optionsToChoose= allCourses ? allCourses.map(eachcourse => optionsList.push({value: eachcourse.name, name:eachcourse.id })) : []
+  const optionsCanSelect= optionsList ? optionsList.map(option => option.name) : [];
+  optionsToChoose =[...optionsList]
+
+  return [
+    {
+      name: "course_id",
+      type: "select",
+      'validation': yup
+      .string().typeError('select something').required('required')
+      .oneOf(
+        optionsCanSelect,
+        "please select one"
+      ),
+    options: optionsToChoose,
+    customProps: { variant: "outlined", id: "", label: "Select a course", defaultValue: '' },
```

Comment:
Please make sure that outlined as a prop is given within the formbuilder. It shouldn't be passed around everywhere because we are only using outlined components all across.